screen/send/details/hooks/useRecalcFee.js

```diff
@@ -0,0 +1,154 @@
+import { BitcoinUnit } from '../../../../models/bitcoinUnits';
+import currency from '../../../../blue_modules/currency';
+import React, { useEffect } from 'react';
+import { WatchOnlyWallet } from '../../../../class';
+import { AbstractHDElectrumWallet } from '../../../../class/wallets/abstract-hd-electrum-wallet';
+import { MintLayerWallet } from '../../../../class/wallets/mintlayer-wallet';
+
+export const useRecalcFee = ({ wallet, networkTransactionFees, feeRate, utxo, addresses, changeAddress, dumb, feePrecalc, setFeePrecalc, balance }) => {
+  // recalc fees in effect so we don't block render
+  useEffect(() => {
+    function recalcBtcFees() {
+      if (!wallet) return; // wait for it
+      const fees = networkTransactionFees;
+      const changeAddress = getChangeAddressFast();
+      const requestedSatPerByte = Number(feeRate);
+      const lutxo = utxo || wallet.getUtxo();
+
+      const options = [
+        { key: 'current', fee: requestedSatPerByte },
+        { key: 'slowFee', fee: fees.slowFee },
+        { key: 'mediumFee', fee: fees.mediumFee },
+        { key: 'fastestFee', fee: fees.fastestFee },
+      ];
+      const newFeePrecalc = { ...feePrecalc };
+
+      for (const opt of options) {
+        let targets = [];
+        for (const transaction of addresses) {
+          if (transaction.amount === BitcoinUnit.MAX) {
+            // single output with MAX
+            targets = [{ address: transaction.address }];
+            break;
+          }
+          const value = parseInt(transaction.amountInCoins);
+          if (value > 0) {
+            targets.push({ address: transaction.address, value });
+          } else if (transaction.amount) {
+            if (currency.btcToSatoshi(transaction.amount) > 0) {
+              targets.push({ address: transaction.address, value: currency.btcToSatoshi(transaction.amount) });
+            }
+          }
+        }
+
+        // if targets is empty, insert dust
+        if (targets.length === 0) {
+          targets.push({ address: '36JxaUrpDzkEerkTf1FzwHNE1Hb7cCjgJV', value: 546 });
```

Comment:
What is that addresses? Is that refactoring from the bluewallet part?

Reply:
Right, It's refactoring from bluewallet part, I just moved it to hook.